build/nightly-build/job-template.yml

```diff
@@ -0,0 +1,116 @@
+parameters:
+  name: ''
+  architecture: x64
+  buildScript: ''
+  pool: {}
+  customMatrixes: ''
+  container: ''
+
+jobs:
+  - job: ${{ parameters.name }}
+    timeoutInMinutes: 20
+    variables:
+      dotnetPath: $(Build.SourcesDirectory)/Tools/dotnetcli/dotnet
+      nugetFeed: https://pkgs.dev.azure.com/dnceng/public/_packaging/MachineLearning/nuget/v3/index.json
+      nightlyBuildProjPath: $(Build.SourcesDirectory)/test/Microsoft.ML.NightlyBuild.Tests/Microsoft.ML.NightlyBuild.Tests.csproj
+      nightlyBuildRunPath: (Build.SourcesDirectory)/bin/AnyCPU.(_configuration)/Microsoft.ML.NightlyBuild.Tests/netcoreapp2.1
+      packageUpdaterProjPath: $(Build.SourcesDirectory)/test/Microsoft.ML.NugetPackageVersionUpdater/Microsoft.ML.NugetPackageVersionUpdater.csproj
+      versionFilePath: $(Build.SourcesDirectory)/test/Microsoft.ML.NugetPackageVersionUpdater/latest_versions.txt
+    strategy:
+      matrix:
+        ${{ if eq(parameters.customMatrixes, '') }}:
+          Debug_Build:
+            _configuration: Debug
+            _config_short: D
+            _includeBenchmarkData: false
+          Release_Build:
+            _configuration: Release
+            _config_short: R
+            _includeBenchmarkData: true
+        ${{ if ne(parameters.customMatrixes, '') }}:
+          ${{ insert }}: ${{ parameters.customMatrixes }}
+
+    pool: ${{ parameters.pool }}
+    ${{ if ne(parameters.container, '') }}:
+      container: ${{ parameters.container }}
+
+    steps:
+    - ${{ if eq(parameters.pool.name, 'Hosted macOS High Sierra') }}:
+      - script: brew update && brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/f5b1ac99a7fba27c19cee0bc4f036775c889b359/Formula/libomp.rb && brew install mono-libgdiplus gettext && brew link gettext --force && brew link libomp --force
+        displayName: Install build dependencies
+    - ${{ if eq(parameters.name, 'Ubuntu_x64_NetCoreApp21') }}:
+      - bash: echo "##vso[task.setvariable variable=LD_LIBRARY_PATH](nightlyBuildRunPath):LD_LIBRARY_PATH"
+        displayName: Set LD_LIBRARY_PATH for Ubuntu to locate Native shared library
+    - script: parameters.buildScript−(_configuration) -buildArch=${{ parameters.architecture }}
```

Comment:
Please prepend variables with `$` to be consistent... `(configuration)` -> `$(configuration)`

Also in the rest of this file.

Reply:
Yes, will do

---
In reply to: [343901037](https://github.com/dotnet/machinelearning/pull/4444#discussion_r343901037) [](ancestors = 343901037)